packages/typo3-docs-theme/src/TextRoles/PhpTextRole.php

```diff
@@ -21,6 +26,48 @@ public function getAliases(): array
 
     public function processNode(DocumentParserContext $documentParserContext, string $role, string $content, string $rawContent): InlineNode
     {
+        $fqn = [];
+        $rawContent = trim($rawContent);
+        if (str_contains($rawContent, '\\') && $this->fullyQualifiedNameService->isFullyQualifiedName($rawContent, $fqn)) {
+            if (!str_starts_with($rawContent, '\\')) {
+                $rawContent = '\\' . $rawContent;
+            }
+            return new CodeInlineNode($rawContent, 'PHP class', $rawContent);
+        }
+        if (str_contains($rawContent, '::') || str_contains($rawContent, '->')) {
+            new CodeInlineNode($rawContent, 'PHP method', $rawContent);
+        }
+        if (str_starts_with($rawContent, '$GLOBALS[\'TYPO3_CONF_VARS\']')) {
+            return new CodeInlineNode(
+                $rawContent,
+                'Global PHP configuration',
+                'The main configuration is achieved via a set of global
+                settings stored in a global array called $GLOBALS[\'TYPO3_CONF_VARS\'].
+                They are commonly set in config/system/settings.php, config/system/additional.php,
+                or the ext_localconf.php file of an extension. '
+            );
+        }
+        if (str_starts_with($rawContent, '$GLOBALS[\'TCA\']')) {
+            return new CodeInlineNode(
+                $rawContent,
+                'Table Configuration Array',
+                'The TCA - Table Configuration Array - is a layer on
+                top of the database tables that TYPO3 can operate on.
+                It should be defined within the folder Configuration/TCA in an
+                extension.
+                '
+            );
+        }
+        if (str_starts_with($rawContent, '$GLOBALS[\'TSFE\']')) {
+            return new CodeInlineNode(
+                $rawContent,
+                'TypoScriptFrontendController',
+                'TSFE is short for \TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController,
+                    a class which exists in the system extension EXT:frontend.
+                    It is available as global array $GLOBALS[\'TSFE\'] in PHP.
```

Comment:
It is also available as PSR-7 request attribute frontend.controller in frontend (which should be preferred instead of the global variable) - but only for newer versions. I assume, detecting the version is not possible?

Reply:
We will output this help text, wherever a string in `:php:` tag starts with $GLOBALS['TSFE']. So when the global variable is not mentioned anymore, the help text will not be displayed anymore. Or only in the changelog for old versions.